ADLStream/evaluation/base_evaluator.py

```diff
@@ -167,9 +182,11 @@ def run(self, context):
             new_results, instances = self.evaluate()
             if new_results:
                 self.write_results(new_results, instances)
+                self.write_predictions()
```

Comment:
Please remove `write_predictions` from here. 
As discussed, it should go (only) in the update prediction function:
```python
def update_predictions(self, context):
      x, y, o = context.get_predictions()
      self.x_eval += x
      self.y_eval += y
      self.o_eval += o
      self.write_predictions(o)
```